### PR TITLE
OSD-24348: Update base image to ubi9

### PR DESCRIPTION
**Pull request by Dee-6777**
Update CI YAML to use ubi9 base image

- Changed base image from ubi8 to ubi9 in CI configuration
- Resolves glibc version mismatch in CI pipeline
- Aligns with the latest system requirements and dependencies
- Ensures compatibility with oc binary and updated dependencies during CI tests

### CODE REVIEW

File: ci-operator/config/openshift/backplane-cli/openshift-backplane-cli-main.yaml [hunk 1]
```diff
@@ -2,12 +2,12 @@ base_images:
...
     name: ubi
     namespace: ocp
-    tag: "8"
+    tag: "9"
 build_root:
   image_stream_tag:
     name: release
     namespace: openshift
-    tag: rhel-8-release-golang-1.21-openshift-4.16
+    tag: rhel-9-release-golang-1.21-openshift-4.16
```
Comment (feichashao) [inline ci-operator/config/openshift/backplane-cli/openshift-backplane-cli-main.yaml:10]: Hi @Dee-6777 , given that I don't have much context on this repo, could you help me understand where these tags come from and how they work?

Comment (Dee-6777) [reply ci-operator/config/openshift/backplane-cli/openshift-backplane-cli-main.yaml:10]: The tag refers to a specific version of the ubi image, such as "8" or "9". For instance, tag: 8 refers to the latest available ubi 8 image, while tag: 9 would reference the latest available ubi9.
When changed to 9, CI will pull the corresponding ubi9 image for the build.

**Comment by feichashao** [comment]
Hey @Dee-6777 , for the build image, we can use `rhel-9-release-golang-1.21-openshift-4.16` , from the name it is RHEL9 with go 1.21, so it should fit the current go version of bp-cli.
```
$ skopeo list-tags docker://registry.ci.openshift.org/openshift/release | grep golang
rhel-9-release-golang-1.21-openshift-4.16
```

**Pull request merged**